### PR TITLE
[#6997] Add admin keyword support to purge_cache (4-2-stable)

### PR DESCRIPTION
**Pull request by alanking**
Once again, and perhaps finally, addresses #6997 

Tests exist in companion PR: https://github.com/irods/irods_resource_plugin_s3/pull/2106

One thing you may notice is that `purge_cache` is a bit of a misnomer... The function actually just trims whatever replica you specify with the passed in `DataObjInfo&`. This makes it a pretty handy replacement to calling trim API directly. I renamed it to `trim_replica` since it's a server-only function. I left this in a separate commit so it can be easily reverted.

Unit tests passed. Running regular test suite to make sure nothing fell over. Working with @JustinKyleJames to check the Glacier tests.

### CODE REVIEW

File: server/core/include/finalize_utilities.hpp [hunk 1]
```diff
@@ -91,8 +91,14 @@ namespace irods
     /// \since 4.2.9
     auto apply_static_post_pep(RsComm& _comm, l1desc& _l1desc, const int _operation_status, std::string_view _pep_name) -> int;
 
-    // TODO: ...remove this.
-    auto purge_cache(RsComm& _comm, DataObjInfo& _info) -> int;
+    /// \brief Trims the replica referred to by \p _info.
+    ///
+    /// \param[in] _comm
```
Comment (korydraughn) [inline server/core/include/finalize_utilities.hpp:96]: Missing description for `_comm`.

**Comment by korydraughn** [comment]
Other than the comment about documentation, everything looks good.

File: server/core/include/finalize_utilities.hpp (hunk 1 above)
Comment (alanking) [reply server/core/include/finalize_utilities.hpp]: Done. Let me know if that's sufficient.

Comment (korydraughn) [reply server/core/include/finalize_utilities.hpp]: That's good.

**Comment by korydraughn** [comment]
Waiting for test results before approving.

**Comment by alanking** [comment]
Core tests passed

**Review by korydraughn — APPROVED** [review]
Squash, pound, and merge when ready.

**Pull request merged**